x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/SentenceBoundaryChunker.java

```diff
@@ -66,6 +66,10 @@ public List<String> chunk(String input, ChunkingSettings chunkingSettings) {
      * @return The input text chunked
      */
     public List<String> chunk(String input, int maxNumberWordsPerChunk, boolean includePrecedingSentence) {
+        if (input.isEmpty()) {
+            return List.of("");
+        }
```

Comment:
Do we need to trim the input before checking if it's empty? How does the chunker handle input that is only whitespace?

Reply:
Good catch @Mikep86 thanks. 

There's a whole class of bugs for "things that don't chunk". Pure whitespace, a single character or the same character repeated thousands of times don't chunk. I've added test cases for these situations and the solution I've implemented here in this is to return the original input if it did not chunk.  This applies to both the Word and Sentence chunker.

This makes me wonder if there should be an upper limit on the chunk size in terms of number of characters. A badly formed input contain latin characters but no whitespace would result in a single large chunk, think a binary file base64 encoded.